### PR TITLE
AWS AMI Generate AIO Configuration

### PR DESCRIPTION
**Pull request by travelton**
### Description
These changes support a new All-In-One configuration (auth, proxy, and node on a single instance) option, which will be utilized by our Terraform examples (coming in a second PR). 

Additionally, a few changes to the README to describe how to build the AMI for development purposes.

_Note: We may need to upgrade Packer on Jenkins (v1.4.0+). The template method `clean_ami_name` has been renamed to `clean_resource_name` in the latest Packer versions._

### CODE REVIEW

**Review by russjones — APPROVED** [review]
Looks good to me, but @webvictim would be the person to check with.

File: assets/marketplace/README.md [hunk 1]
```diff
@@ -24,23 +28,28 @@ Follow instructions at: https://www.packer.io/docs/install/index.html
...
+4. Set the following Makefile values:
 
-After stack has been provisioned, login to the AWS Console and get the IP address of the server.
+| Param               | Description                                                                                                |
+|---------------------|------------------------------------------------------------------------------------------------------------|
+| BUILD_VPC_ID        | With the region you selected in step 3, create or use an existing VPC. ex. `vpc-xxxxxxxx`.                 |
+| BUILD_SUBNET_ID     | Within the VPC above, select a subnet. ex. `subnet-xxxxxxxx`                                               |
+| AWS_REGION          | Region you selected in step 3. ex. `us-east-1`                                                             |
+| TELEPORT_VERSION    | Teleport version. See [Teleport releases](https://github.com/gravitational/teleport/releases). ex. `4.2.9` |
+| INSTANCE_TYPE       | The instance type used for the build. ex. `ts.micro`                                                       |
```
Comment (webvictim) [inline assets/marketplace/README.md:42]: t2.micro

File: assets/marketplace/README.md [hunk 2]
```diff
@@ -24,23 +28,28 @@ Follow instructions at: https://www.packer.io/docs/install/index.html
...
+make oss
 ```
-ssh -i key.pem ec2-user@$SERVER_IP
-
-# For OSS
-
-sudo -u teleport tctl users add bob bob
 
-# For Enterprise
+6. Once complete, your AMI should be available, in the regions you specified, with the name  `teleport-debug-ami-<type>-<version>`.
```
Comment (webvictim) [inline assets/marketplace/README.md:51]: Add an example? "e.g. `teleport-debug-ami-oss-4.2.9`"

**Review by webvictim — APPROVED** [review]
Looks OK to me with a couple of nits.

**Comment by webvictim** [comment]
ok to test

**Comment by webvictim** [comment]
retest this please

**Comment by travelton** [comment]
@webvictim Do you foresee any issues with the Packer changes, `clean_ami_name` to `clean_resource_name`? The machine that uses Packer to build AMIs may need to be updated, but I'm not sure where those dependencies are defined.

**Comment by webvictim** [comment]
@travelton It looks like `clean_resource_name` was introduced in v1.4.0 so it'll work fine with the version that Jenkins is using. I have the same version installed locally and it worked fine when I tested it.
```
[centos@jenkins ~]$ packer version
Packer v1.4.5
```

**Comment by webvictim** [comment]
@travelton It looks like `clean_resource_name` was introduced in v1.4.0 so it'll work fine with the version that Jenkins is using. I have the same version installed locally and it worked fine when I tested it.
```
[user@jenkins ~]$ packer version
Packer v1.4.5
```

**Pull request merged**